dace/sdfg/sdfg.py

```diff
@@ -2047,18 +2100,35 @@ def add_datadesc(self, name: str, datadesc: dt.Data, find_new_name=False) -> str
             if name in self._pgrids:
                 raise FileExistsError(f'Can not create data descriptor "{name}", the name is used by a ProcessGrid.')
 
-        def _add_symbols(sdfg: SDFG, desc: dt.Data):
-            if isinstance(desc, dt.Structure):
-                for v in desc.members.values():
-                    if isinstance(v, dt.Data):
-                        _add_symbols(sdfg, v)
-            for sym in desc.free_symbols:
-                if sym.name not in sdfg.symbols:
-                    sdfg.add_symbol(sym.name, sym.dtype)
-
         # Add the data descriptor to the SDFG and all symbols that are not yet known.
         self._arrays[name] = datadesc
-        _add_symbols(self, datadesc)
+        self._add_symbols(datadesc)
+
+        if (
+            datadesc.transient is True and
+            isinstance(datadesc, dt.Array) and
+            name != "__return"
+            ):
+            size_desc_name = f"{name}_size"
+            size_desc = dt.Array(dtype=dace.uint64,
+                                shape=(len(datadesc.shape),),
+                                storage=dtypes.StorageType.Default,
```

Comment:
Shouldn't this always be CPU Heap? Default can also be resolved to GPU Global depending on the context, unless special care is taken in code generation and GPU transform. I think you have done so, but maybe it is better to make it clear here too.

Reply:
I have independently noticed this. I decided to go with CPU_Heap. (But the codegen always allocates size arrays on CPU_Heap. I did not go with Registers due to the dangers of them considered accessible in GPU kernels).